compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/PromotionImpls.cpp

```diff
@@ -111,14 +112,35 @@ Value cacheSwizzlePromotionImpl(OpBuilder &builder, OpOperand &operand,
   }
 
   Location loc = promotedValue.getLoc();
-  // Use the size of the inner most dimension as the cache swizzle value.
-  // This is a very rudimentary choice, but functions well enough as a
+  // Use the size in bytes of the inner most dimension as the cache swizzle
+  // value. This is a very rudimentary choice, but functions well enough as a
   // default.
-  Value cacheSwizzleVal = getValueOrCreateConstantIndexOp(
+  AffineExpr s0, s1;
+  bindSymbols(builder.getContext(), s0, s1);
+  Value dtype = getValueOrCreateConstantIntOp(
```

Comment:
Since you're already creating an op with `arith::ConstantIndexOp::create` you don't need to wrap it in `getValueOrCreateConstantIntOp`.